src/main/src/test/java/org/geoserver/catalog/ResourcePoolTest.java

```diff
@@ -343,6 +344,33 @@ public void testDropCoverageStore() throws Exception {
         getGeoServer().reload();
     }
 
+    @Test
+    public void testNativeCoverageName() throws Exception { // GEOS-9236
+        // build the store
+        Catalog cat = getCatalog();
+        CatalogBuilder cb = new CatalogBuilder(cat);
+        CoverageStoreInfo store = cb.buildCoverageStore("geotiff");
+
+        store.setURL(
+                MockData.class
+                        .getResource("/org/geoserver/data/test/tazdem.tiff")
+                        .toExternalForm());
+        store.setType("GeoTIFF");
+        cat.add(store);
+
+        // build the coverage
+        cb.setStore(store);
+        CoverageInfo ci = cb.buildCoverage();
+        ci.setNativeCoverageName("geotiff_coverage");
+        cat.add(ci);
+
+        GridCoverageReader reader = ci.getGridCoverageReader(null, GeoTools.getDefaultHints());
+
+        assertNotNull(ci);
+        assertTrue(reader.getFormat() instanceof GeoTiffFormat);
+        assertEquals("tazdem", ci.getNativeCoverageName());
```

Comment:
I would also add the part of calls involved in the log of the reporting JIRA to making sure we are covering the same scenario.